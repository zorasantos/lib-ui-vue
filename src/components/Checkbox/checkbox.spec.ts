import { mount } from "@vue/test-utils";
import CheckBox from "./Checkbox.vue";

describe("CheckBox", () => {
  test("renders label and input correctly", () => {
    const wrapper = mount(CheckBox, {
      props: {
        label: "Accept term!",
        errorMessages: undefined,
        modelValue: false,
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.find("label").text()).toBe("Accept term!");
    expect(wrapper.find("input").element.checked).toBe(false);
    expect(wrapper.find("label").classes("text-red-500")).toBe(false);
  });

  test("displays error message when errorMessages prop is provided", () => {
    const wrapper = mount(CheckBox, {
      props: {
        label: "Accept term!",
        errorMessages: "Term field is required!",
        modelValue: false,
      },
    });

    expect(wrapper.find("label").classes("text-red-500")).toBe(true);
    expect(wrapper.find("label").classes("text-gray-600")).toBe(false);

    expect(wrapper.text()).toContain("Term field is required!");
    expect(wrapper.find("span").classes("text-red-500")).toBe(true);
  });

  test("emits input event with correct value when input is typed", async () => {
    const wrapper = mount(CheckBox, {
      props: {
        label: "Accept term!",
        errorMessages: undefined,
        modelValue: true,
      },
    });

    // Possivelmente temos que mockar toda a l√≥gica do checkbox

    wrapper.vm.$emit("update:modelValue", true);

    const value = wrapper.emitted("update:modelValue") as Array<[][]>;

    await wrapper.find('input[type="checkbox"]').trigger("click");
    expect(wrapper.find("label").classes("text-gray-600")).toBe(true);
    expect(value[0][0]).toBe(true);
  });
});
