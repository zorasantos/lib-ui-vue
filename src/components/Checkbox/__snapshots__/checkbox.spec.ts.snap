// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CheckBox > renders label and input correctly 1`] = `
"<div class=\\"flex items-center\\"><input id=\\"checkbox\\" type=\\"checkbox\\" class=\\"w-4 h-4 cursor-pointer border-8 text-blue-500 bg-gray-100 rounded\\"><label for=\\"checkbox\\" class=\\"ml-2 block text-base font-medium text-gray-600\\">Accept term!</label></div>
<!--v-if-->"
`;
